modules/types_proto/src/bounds.cpp

```diff
@@ -0,0 +1,45 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#include "hephaestus/types_proto/bounds.h"
+
+#include "hephaestus/types//bounds.h"
+#include "hephaestus/types/proto/bounds.pb.h"
+#include "hephaestus/utils/exception.h"
+
+namespace heph::types::internal {
+
+auto getAsProto(const BoundsType& bounds_type) -> proto::BoundsType {
+  switch (bounds_type) {
+    case BoundsType::INCLUSIVE:
+      return proto::BoundsType::INCLUSIVE;
+    case BoundsType::LEFT_OPEN:
+      return proto::BoundsType::LEFT_OPEN;
+    case BoundsType::RIGHT_OPEN:
+      return proto::BoundsType::RIGHT_OPEN;
+    case BoundsType::OPEN:
+      return proto::BoundsType::OPEN;
+    default:
+      throwException<InvalidParameterException>("Invalid bounds type");
+      return {};  // unreachable
```

Comment:
one way to get away with not doing that is to unit test the alignment of enums both in proto and c++ types...The unit test can be macro'd and automated. wdyt?

Reply:
Maybe we mean different things with aligned. I meant that the underlying int values are the same, which protobuf does not allow AFAIK, given it starts at 0 and monotonously counts up.

In our source code, I for example use a scaler which maps enum values as follows
```
FOO_40_HZ = 1, FOO_20_HZ = 2, FOO_1_HZ = 40
```

If you simply mean that there exists a 1:1 mapping between all enum values, 100% agreed. We'll enforce this soon!